observer/probers/crl/crl_conf.go

```diff
@@ -0,0 +1,152 @@
+package probers
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	nextUpdateName = "obs_crl_next_update"
+	thisUpdateName = "obs_crl_this_update"
+	certCountName  = "obs_crl_revoked_cert_count"
+)
+
+// CRLConf is exported to receive YAML configuration
+type CRLConf struct {
+	URL string `yaml:"url"`
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Configurer`.
+func (c CRLConf) Kind() string {
+	return "CRL"
+}
+
+// UnmarshalSettings constructs a CRLConf object from YAML as bytes.
+func (c CRLConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf CRLConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c CRLConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
+	}
+	if url.Scheme == "" {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, missing scheme", c.URL)
+	}
+	return nil
+}
+
+func (c CRLConf) validateCollectors(colls map[string]prometheus.Collector) (*prometheus.GaugeVec, *prometheus.GaugeVec, *prometheus.GaugeVec, error) {
+	if colls == nil {
+		message := "crl prober defines metrics but received nil collector map"
+		return nil, nil, nil, errors.New(message)
+	}
+	var nu, tu, rcc *prometheus.GaugeVec
+	for name, coll := range colls {
+		switch name {
+		case nextUpdateName:
+			_, ok := coll.(*prometheus.GaugeVec)
+			if !ok {
+				message := fmt.Sprintf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, nu)
+				return nil, nil, nil, errors.New(message)
+			}
+			nu = coll.(*prometheus.GaugeVec)
+		case thisUpdateName:
+			_, ok := coll.(*prometheus.GaugeVec)
+			if !ok {
+				message := fmt.Sprintf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, tu)
+				return nil, nil, nil, errors.New(message)
+			}
+			tu = coll.(*prometheus.GaugeVec)
+		case certCountName:
+			_, ok := coll.(*prometheus.GaugeVec)
+			if !ok {
+				message := fmt.Sprintf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, rcc)
+				return nil, nil, nil, errors.New(message)
+			}
+			rcc = coll.(*prometheus.GaugeVec)
+		default:
+			message := fmt.Sprintf("crl prober received unexpected collector '%s'", name)
+			return nil, nil, nil, errors.New(message)
+		}
```

Comment:
If you pull all of this logic directly into `MakeProber()` then you can initialize a `CRLProbe`  right at the top and assert them directly into their destination fields in that `CRLProbe`.

Reply:
Then, if you want to keep validation separate you can always pass the prober through and emit an error.

Reply:
Done! I moved some of the logic, then decided I didn't like doing only the last bits of validation in a separate function, so all of this logic has been moved into `MakeProber()`